### ISSUE TITLE
DROCC Learning Scheduler

### ISSUE THREAD

**Issue by mpierrau** [SachinG007/DROCC_icml2020#4]
Hi!

I am attempting to reproduce your results in the DROCC paper for my master's thesis and I was wondering if you could specify which learning rate scheduler was used during the experiments and the settings?

From the code in Microsoft's EdgeML [repo](https://github.com/microsoft/EdgeML/blob/master/pytorch/edgeml_pytorch/trainer/drocc_trainer.py) it is not clear what type the scheduler is expected to be. It is not specified in the parameter description.
From the looks of it, it doesn't seem to be a regular torch.optim.lr_scheduler, as the arguments differ (epoch, total_epochs, only_ce_epochs, learning_rate, self.optimizer) from a "regular" lr_scheduler, which has optimizer as first positional argument. The scheduler also is not updated at the end of each epoch in the code.

in this repo the following snippet is found in cifar/code/train_cifar_pgd.py:

```
#LR SCHEDULER
def adjust_learning_rate(optimizer, epoch, args):
    """decrease the learning rate"""
    lr = args.lr
    if epoch <= args.epochs:
        lr = args.lr * 0.01
    if epoch <= 0.80*args.epochs:
        lr = args.lr * 0.1 
    if epoch <= 0.40*args.epochs:
        lr = args.lr    
    for param_group in optimizer.param_groups:
        param_group['lr'] = lr
```

which hints at the used learning rate scheduler for the CIFAR-10 experiment. The can easily be replaced by a torch.lr_scheduler.StepLR scheduler with step_size=0.4*n_epochs and gamma=0.1, where n_epochs is the total number of epochs.

However, the arguments for lr_scheduler in drocc_trainer.py in the EdgeML repo suggests that the learning rate may be different for the "pretraining" (without adversarial loss) phase.

I suppose I can experiment a bit with it, but I definitely think that the lr_scheduler utilized in drocc_trainer.py should be either removed or properly explained, and I would much appreciate some clarification on the parameters used during the experiments.

Cheers!
